after_both.go

```diff
@@ -0,0 +1,32 @@
+package asynctask
+
+import "context"
+
+// AfterBothFunc is a function that has 2 input.
+type AfterBothFunc[T, S, R any] func(context.Context, *T, *S) (*R, error)
+
+// AfterBoth runs the function after both 2 input task finished, and will be feed with result from 2 input task.
+func AfterBoth[T, S, R any](ctx context.Context, tskT *Task[T], tskS *Task[S], next AfterBothFunc[T, S, R]) *Task[R] {
+	return Start(ctx, func(fCtx context.Context) (*R, error) {
+		t, err := tskT.Result(fCtx)
+		if err != nil {
+			return nil, err
+		}
+
+		s, err := tskS.Result(fCtx)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
This suggests error from `taskT` always take priority as error.  If this is the expected behavior, it should be clearly stated.  Also,  when `taskT` errors out, `taskS` could still be running.  This is not truly **After Both**.

Reply:
1. if one inputTask failed, this afterBoth task should fail anyway, right?
2. until later we introduce some option to allow ignore partial error, or we still fail, but wait for both task.

anyway, I will updade comments

Reply:
we have WaitAllOptions in wait_all.go, we could reuse here in this afterBoth in future PRs, and then adjust the behavior according based on the option.